### PR TITLE
[TIMOB-24567] iOS: Use Swift 3.1 for watchOS templates, update pairing table

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24567

ioslib@1.3.3: https://github.com/appcelerator/ioslib/pull/56

### CODE REVIEW

File: node_modules/ioslib/lib/xcode.js [hunk 1]
```diff
@@ -30,42 +30,30 @@ var cache,
...
-			'2.2': true, // watchOS 2.2
-			'3.x': true  // watchOS 3.x
-		},
-		'10.x': {        // iOS 10.x
-			'2.2': true, // watchOS 2.2
-			'3.x': true  // watchOS 3.x
+	'>=8.3': {                      // Xcode 8.3
+		'10.x': {                   //   - iOS 10.x
+			'>=3.x': true           //   - watchOS 3.x
 		}
```
Comment (hansemannn) [inline node_modules/ioslib/lib/xcode.js]: @cb1kenobi As soon as we conclude on the table, I'd do an ioslib PR.

**Comment by cb1kenobi** [comment]
This is awesome! Thanks for working on this.

Don't bother updating the ioslib files in titanium_mobile for other than testing. You are correct, we need to do them ioslib, publish a release, and update ioslib in titanium_mobile.

File: node_modules/ioslib/lib/xcode.js [hunk 2]
```diff
@@ -315,9 +303,9 @@ exports.detect = function detect(options, callback) {
 						}
 					};
 
-					Object.keys(simulatorDevicePairCompatiblity).some(function (xcodeRange) {
+					Object.keys(simulatorDevicePairCompatibility).some(function (xcodeRange) {
```
Comment (cb1kenobi) [inline node_modules/ioslib/lib/xcode.js]: lol

**Comment by cb1kenobi** [comment]
We need to stop modifying the "watchos2-swift" project template and create a "watchos3-swift".

**Comment by hansemannn** [comment]
Agree! Need to check where to edit the references, but yeah.

**Comment by hansemannn** [comment]
Updated the template. Is it possible that we don't even use the `watchos2-objc` template anymore? The only reference I can find is https://github.com/appcelerator/titanium_mobile/pull/8952/commits/684a91e6678df59144206a3de90d8e794d7b1eeb#diff-d69717379468fe83d78fc685c3797eefR68.

**Comment by cb1kenobi** [comment]
Yeah, I think that's the only place you need to specify the template name.

**Comment by hansemannn** [comment]
Then w should drop the watchos2-objc template completely. When we create a new watchOS extension, it will use the Swift template anyway.

**Comment by cb1kenobi** [comment]
Works like a champ! **APPROVED**

**Pull request merged**